### DISCUSSION TITLE
Modbus Zähler einrichten (VSM-102)

### DISCUSSION THREAD

**Discussion by bebu259** [camueller/SmartApplianceEnabler#473] [Installation und Betrieb]
Hallo Axel,

ich muss doch noch mal eine Frage stellen... 🫢

Ich versuche einem **Volcraft VSM-102** Drehstromzähler per Modbus einzurichten. Die Dokumentation gibt diese Infos:

<img width="611" alt="Bildschirmfoto 2023-02-14 um 14 54 01" src="https://user-images.githubusercontent.com/1553063/218758405-029df616-b7d6-42ec-8879-59cdc460a2f6.png">

Jetzt versuche ich wie in der Doku unter [**Überprüfung von Modbus/TCP**](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Modbus_DE.md#%C3%BCberpr%C3%BCfung-modbustcp) beschrieben mit `mbpoll` Kontakt zum Zähler zu bekommen. 

Laut Zähler-Dokumentation soll `/?!<CR><LF>` an den Zähler gesendet werden:

| Code | hex | dezimal |
|:-----:|:----:|:------
|  `/`           | 2F |  47
|  `?`          | 3F |  63
|  `!`           | 21 |  33 
| `<CR>`   | 0D | 13
| `<LF>`    | 0A | 10

Ich habe diese Anfrage gebaut:
```
mbpoll /dev/ttyUSB0 47 63 33 13 10 -b 9600 -d 7 -P even -s 1 -v
```

Als Antwort bekommen ich:
```
sae@raspberrypi:~ $ mbpoll /dev/ttyUSB0 47 63 33 13 10 -b 9600 -d 7 -P even -s 1 -v
debug enabled
Set mode to RTU for serial port
Set device=/dev/ttyUSB0
5 write data have been found
Set data=47
Word[0]=0x2F
Set data=63
Word[1]=0x3F
Set data=33
Word[2]=0x21
Set data=13
Word[3]=0xD
Set data=10
Word[4]=0xA
mbpoll 1.4-12 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright © 2015-2019 Pascal JEAN, https://github.com/epsilonrt/mbpoll
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; type 'mbpoll -w' for details.

Opening /dev/ttyUSB0 at 9600 bauds (E, 7, 1)
Set response timeout to 1 sec, 0 us
Protocol configuration: Modbus RTU
Slave configuration...: address = [1]
                        start reference = 1, count = 5
Communication.........: /dev/ttyUSB0,       9600-7E1
                        t/o 1.00 s, poll rate 1000 ms
Data type.............: 16-bit register, output (holding) register table

[01][10][00][00][00][05][0A][00][2F][00][3F][00][21][00][0D][00][0A][BB][09]
Waiting for a confirmation...
ERROR Connection timed out: select
Write output (holding) register failed: Connection timed out
```

Ich kenne `mbpoll` leider nicht gut, aber diese Zeile ist vermutlich das, was an den Bus geschickt wird:

<img width="567" alt="Bildschirmfoto 2023-02-14 um 14 57 06" src="https://user-images.githubusercontent.com/1553063/218759529-9ab54b90-20d0-40f7-9dc6-ab6fe7da7af3.png">

Darin ist der Anfragetext zwar enthalten, aber auch vieles anderes...

Hast du einen Tip, ob ich hier auf dem richtigen Weg bin?

Geht es überhaupt, einen Zähler mit `mbpoll` abzufragen, wenn das in zwei Schritten (wie beim VSM-102) passieren muss?

**Comment by camueller** [comment]
Das scheint kein richtiges Modbus-Protokoll zu sein. Ich kann Dir leider nicht weiterhelfen mit diesem Zähler.

**Reply by bebu259** [reply]
Das hatte ich schon vermutet, dass es gar kein Modbus Zähler ist.
Ich werde mich nach einem anderen Zähler umschauen...
Danke!

**Comment by camueller** [comment]
Ich habe mit den Zählern von https://stromzähler.eu sehr gute Erfahrungen gemacht.

**Reply by bebu259** [reply]
Danke für den Tip! 
Ich habe jetzt einen ‎**SDM72D-M-MID V2** bestellt.

**Comment by camueller** [comment]
Den habe ich auch (ohne MID) für die Wallbox.

**Reply by bebu259** [reply]
Mit dem **SDM72D** lief die Installation problemlos!

**Reply by camueller** [reply]
Sehr gut.
